### PR TITLE
merge PR 15026 15051 15067 15070 from v0.79.HERO-rc into master

### PR DESCRIPTION
**Pull request by roxanneskelly**
https://highfidelity.manuscript.com/f/cases/21583/merge-PR-15026-15051-15069-15070-from-v0-79-HERO-rc-into-master

### CODE REVIEW

**Comment by SimonWalton-HiFi** [comment]
This looks to have all the avatar-priority changes in.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17892/HighFidelity-Beta-PR15104-fae6393.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17892/HighFidelity-Beta-Interface-PR15104-fae6393.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17892/HighFidelity-Beta-PR15104-fae6393.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17892/HighFidelity-Quest-Beta-PR15104-fae6393.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17892/HighFidelity-Beta-PR15104-fae6393.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17892/HighFidelity-Beta-Interface-PR15104-fae6393.exe

**Comment by howard-stearns** [comment]
The description says this includes 15069. I assume that's a typo for 15067?

File: libraries/entities/src/ZoneEntityItem.cpp [hunk 1]
```diff
@@ -211,6 +214,7 @@ EntityPropertyFlags ZoneEntityItem::getEntityProperties(EncodeBitstreamParams& p
     requestedProperties += PROP_FLYING_ALLOWED;
     requestedProperties += PROP_GHOSTING_ALLOWED;
     requestedProperties += PROP_FILTER_URL;
+    requestedProperties += PROP_AVATAR_PRIORITY;
```
Comment (SimonWalton-HiFi) [inline libraries/entities/src/ZoneEntityItem.cpp]: Should move this to just after `requestedProperties += PROP_BLOOM_MODE;`, to be consistent.

File: libraries/entities/src/EntityItem.cpp [hunk 2]
```diff
@@ -41,6 +41,7 @@
 #include "EntitySimulation.h"
 #include "EntityDynamicFactoryInterface.h"
 
+//#define WANT_DEBUG
```
Comment (SimonWalton-HiFi) [inline libraries/entities/src/EntityItem.cpp]: May as well take this out.

File: libraries/entities/src/ZoneEntityItem.cpp [hunk 3]
```diff
@@ -269,6 +274,7 @@ void ZoneEntityItem::debugDump() const {
     qCDebug(entities) << "   _ambientLightMode:" << EntityItemProperties::getComponentModeAsString(_ambientLightMode);
     qCDebug(entities) << "         _skyboxMode:" << EntityItemProperties::getComponentModeAsString(_skyboxMode);
     qCDebug(entities) << "          _bloomMode:" << EntityItemProperties::getComponentModeAsString(_bloomMode);
+    qCDebug(entities) << "     _avatarPriority:" << getAvatarPriority();
```
Comment (SimonWalton-HiFi) [inline libraries/entities/src/ZoneEntityItem.cpp]: Probably should be `EntityItemProperties::getAvatarPriorityAsString()` but we can leave it for the next round.

**Pull request merged**